File: api/go/models/match.go
package models

import "go/helpers"

type Match struct {
	TeamA           Team `json:"team_a"`
	TeamB           Team `json:"team_b"`
	ScoreDifference int  `json:"score_difference"`
}

func NewMatch(teamA Team, teamB Team) Match {
	scoreDifference := helpers.Abs(teamB.Score - teamA.Score)
	return Match{TeamA: teamA, TeamB: teamB, ScoreDifference: scoreDifference}
}

File: api/node/src/models/Match.ts
import Team from "./Team";

export default class Match {
    team_a: Team;
    team_b: Team;
    score_difference: number;

    constructor(team_a: Team, team_b: Team) {
        this.team_a = team_a
        this.team_b = team_b
        this.score_difference = Math.abs(team_a.score - team_b.score)
    }
}
